plugins/destination/mysql/client/read.go

```diff
@@ -2,83 +2,197 @@ package client
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Int8Builder).Append(int8(v.Int64))
+			}
+		case *arrow.Int16Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Int16Builder).Append(int16(v.Int64))
+			}
+		case *arrow.Int32Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Int32Builder).Append(int32(v.Int64))
+			}
+		case *arrow.Int64Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Int64Builder).Append(v.Int64)
+			}
+		case *arrow.Uint8Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Uint8Builder).Append(uint8(v.Int64))
+			}
+		case *arrow.Uint16Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Uint16Builder).Append(uint16(v.Int64))
+			}
+		case *arrow.Uint32Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Uint32Builder).Append(uint32(v.Int64))
+			}
+		case *arrow.Uint64Type:
+			v := val.(*sql.NullInt64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Uint64Builder).Append(uint64(v.Int64))
+			}
+		case *arrow.Float32Type:
+			v := val.(*sql.NullFloat64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Float32Builder).Append(float32(val.(*sql.NullFloat64).Float64))
+			}
+		case *arrow.Float64Type:
+			v := val.(*sql.NullFloat64)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.Float64Builder).Append(val.(*sql.NullFloat64).Float64)
+			}
+		case *arrow.StringType:
+			v := val.(*sql.NullString)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.StringBuilder).Append(val.(*sql.NullString).String)
+			}
+		case *arrow.LargeStringType:
+			v := val.(*sql.NullString)
+			if !v.Valid {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.LargeStringBuilder).Append(val.(*sql.NullString).String)
+			}
+		case *arrow.BinaryType:
+			if *val.(*[]byte) == nil {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.BinaryBuilder).Append(*val.(*[]byte))
+			}
+		case *arrow.TimestampType:
+			asTime := val.(**time.Time)
+			if *asTime == nil {
+				recordBuilder.Field(i).AppendNull()
+			} else {
+				recordBuilder.Field(i).(*array.TimestampBuilder).Append(arrow.Timestamp((*asTime).UnixMicro()))
+			}
+		case *types.UUIDType:
```

Comment:
This is not needed I believe as the `default` case should take care of that.

Reply:
I think using the default only works if we use the string representation for UUID